### PR TITLE
boards: bl602evb flash and debug notes

### PR DESCRIPTION
**Pull request by btashton**
## Summary
Add more friendly notes for getting started with the bl602evb board.  Including both flashing and using the debugger.

### CODE REVIEW

**Comment by btashton** [comment]
@liangzhanggb and @acassis you both are probably interested.

@liangzhanggb  I saw that there are some notes about using the stub flash loader in the BL flash_tools repo, but I was not quite clear if I could use just use this to then allow GDB to flash the part, so I just documented what I have been using for my development.

File: boards/risc-v/bl602/bl602evb/README.txt [hunk 1]
```diff
@@ -1,10 +1,15 @@
+# Getting Started with nsh configuration
 1. Download and install toolchain
 
   $ curl https://static.dev.sifive.com/dev-tools/riscv64-unknown-elf-gcc-8.3.0-2019.08.0-x86_64-linux-ubuntu14.tar.gz
 
-2. Download Bouffalo lab flash tools
+2. Download flash tools
```
Comment (acassis) [inline boards/risc-v/bl602/bl602evb/README.txt:6]: 2. Download flash tools (choose option "a" or "b")

File: boards/risc-v/bl602/bl602evb/README.txt [hunk 2]
```diff
@@ -18,11 +23,173 @@
 
 4. Connect bl602 and computer via USB
 
-5. Run flash tools, select the nuttx.bin generated in the previous step in the
-   Firmware bin field, and refer to the document for the settings of the remaining fields.
-   Click the download button to download bin to bl602.
+5. flash
```
Comment (acassis) [inline boards/risc-v/bl602/bl602evb/README.txt:26]: Flashing (choose option "a" or "b" based on previous choice)

**Comment by liangzhanggb** [comment]
> @liangzhanggb and @acassis you both are probably interested.
> 
> @liangzhanggb I saw that there are some notes about using the stub flash loader in the BL flash_tools repo, but I was not quite clear if I could use just use this to then allow GDB to flash the part, so I just documented what I have been using for my development.

1. We have a private download tool that supports uart/openocd/jlink. Many mass-produced customers are using it, so we use a private download tool
2. Because of insufficient resources, we have not do it(GDB to flash) now, maybe this function will be added in the future
Thank very much.

**Comment by acassis** [comment]
> > @liangzhanggb and @acassis you both are probably interested.
> > @liangzhanggb I saw that there are some notes about using the stub flash loader in the BL flash_tools repo, but I was not quite clear if I could use just use this to then allow GDB to flash the part, so I just documented what I have been using for my development.
> 
>     1. We have a private download tool that supports uart/openocd/jlink. Many mass-produced customers are using it, so we use a private download tool
> 
>     2. Because of insufficient resources, we have not do it(GDB to flash) now, maybe this function will be added in the future
>        Thank very much.

@liangzhanggb Yes, using OpenOCD/GDB for flashing is a good idea!

BTW a BL602 user commented in the LinkedIn post, he is interested to help adding WiFi support: https://www.linkedin.com/feed/update/urn:li:activity:6759153916614586368/

**Comment by btashton** [comment]
Thanks @acassis made those updates.  I have a larger tutorial that I am planning on publishing in the next week or so that goes into more details of using this chip with NuttX including some of the recent additions.

@liangzhanggb  Fully understand, I really appreciate all the the work that you and the company have put into supporting this chip in NuttX.  I would be very interested in adding SPI support, but I wanted to check that you had not already started that work.

**Comment by liangzhanggb** [comment]
Thanks @btashton , About spi, We have already started,  I am planning on publishing in the this week. But prepare to support master and no_dma,slave and spi_dma will take some time.

**Comment by btashton** [comment]
Understood 👍. For now I'll just help with testing and review.  If there is anything in particular I can help with please let me know.

**Pull request merged**